### DISCUSSION TITLE
Hayward Aquarite+ SV (AQR-PLUS-SV33ST) + Hayward SuperPump SP2622XW251 + Home Assistant

### DISCUSSION THREAD

**Discussion by alexdelprete** [tagyoureit/nodejs-poolController#805] [Q&A]
Hi,

first of all thank you @tagyoureit / @rstrouse for creating and supporting this fantastic solution, I've been reading since yesterday the wiki to learn as much as possible, but I need some advice to have a proper start. :)

I ordered the products in title, the chlorinator has the pH controller included, but I didn't buy the REDOX controller option (for budget reasons). The pump is not variable-speed.

I use Home Assistant, so I plan to use @crewski's custom component: https://github.com/Crewski/njsPC-HA
I will install nodejs-PC in my homelab (docker probably or a Proxmox LXC, I'll decide later).

For RS-485 communication, I'll use the HF5111B because I already used it in the past for other projects (PV inverter), now it's not used anymore, and it worked beautifully in the past, it supports several serial standards and it has an ethernet port.

So I want to control via HA the chlorinator and the pump, ideally I would also like to "emulate" the automatic redox that the chlorinator supports when buying the kit, by regulating production based on the current level.

Now I have some questions/doubts:

1. RS-485 connection: from the chlorinator manual, I see where I should connect, but the labesl are different respect to the info I found in the issues section or in the discussions. I know I need to provide power to it, so I'll use 12VDC both for the HF5111B and the port of the chlorinator. Can you confirm the connection picture below is ok?
2. Does nodejs-PC support interfacing the RS-485 device via IP:PORT vs /dev/USB mapping?
3. Since I don't use a controller, from what I understood I should add the chlorinator as Nixie, and then I have to specify the model/type, correct?

Thanks for any feedback, the products should arrive in a couple of weeks and can't wait to control it with nodejs-PC. 

![image](https://github.com/tagyoureit/nodejs-poolController/assets/7027842/8a92539b-8af1-43b9-8e29-54a47e950ad2)

![image](https://github.com/tagyoureit/nodejs-poolController/assets/7027842/9715ba3a-c853-4616-b02f-550f8b882ec7)

**Comment by CodeSlinger69** [comment]
I can answer two of your questions.

> * Does nodejs-PC support interfacing the RS-485 device via IP:PORT vs /dev/USB mapping?

For sure. I used njsPC in this mode for about 8 months. I had an Elfin EW11 RS485/WiFi bridge hooked up to my Easytouch RS485 bus. This devices exposed the bus as an IP/Port on my network. The relevant config looks like this. In my case I configured the EW11 to work on 192.168.1.4:9801

![01 comms](https://github.com/tagyoureit/nodejs-poolController/assets/10177469/97fd4452-a632-4d23-9f93-8e747e0f102f)

> * Since I don't use a controller, from what I understood I should add the chlorinator as Nixie, and then I have to specify the model/type, correct?

I'm working trough a transition from using a controller to standalone Nixie right now in [this thread](https://github.com/tagyoureit/nodejs-poolController/discussions/801). And yes, as I just discovered [here](https://github.com/tagyoureit/nodejs-poolController/discussions/804), you do have to add the chlorinator model/type manually into the njsPC config yourself.

**Reply by alexdelprete** [reply]
Thanks for answering. Appreciate it. :)

> The relevant config looks like this. In my case I configured the EW11 to work on 192.168.1.4:9801

I saw the ip:port, but what is the local usb port in the screenshot? A virtual port created locally?

![image](https://github.com/tagyoureit/nodejs-poolController/assets/7027842/a206a7e1-9957-4466-af5e-46367b020290)

> yes, as I just discovered [here](https://github.com/tagyoureit/nodejs-poolController/discussions/804), you do have to add the chlorinator model/type manually into the njsPC config yourself.

ok, I will just have to tinker with it a little bit to understand the logic. :)
